File: eAnimalShelter/eAnimalShelter.Model/Requests/VolunteerActivityInsertRequest.cs
namespace eAnimalShelter.Model.Requests
{
    public class VolunteerActivityInsertRequest
    {
        public string Title { get; set; } = string.Empty;

        public string Description { get; set; } = string.Empty;

        public int LocationId { get; set; }

        public DateTime StartDateTime { get; set; }

        public DateTime EndDateTime { get; set; }

        public int MaxVolunteers { get; set; }
    }
}

File: eAnimalShelter/UI/eanimalshelter_desktop/lib/models/requests/volunteer_activity_insert_request.dart
class VolunteerActivityInsertRequest {
  String title;
  String description;
  int locationId;

  DateTime startDateTime;
  DateTime endDateTime;

  int maxVolunteers;

  VolunteerActivityInsertRequest({
    required this.title,
    required this.description,
    required this.locationId,
    required this.startDateTime,
    required this.endDateTime,
    required this.maxVolunteers,
  });

  Map<String, dynamic> toJson() => {
    "title": title,
    "description": description,
    "locationId": locationId,
    "startDateTime":
        startDateTime.toIso8601String(),
    "endDateTime":
        endDateTime.toIso8601String(),
    "maxVolunteers": maxVolunteers,
  };
}